pkg/network/telemetry/reporters.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-go/v5/statsd"
+)
+
+// common prefix used across all statsd metric
+const statsdPrefix = "datadog.system_probe.network_tracer."
+
+var statsdDelta deltaCalculator
+
+// ReportStatsd flushes all metrics tagged with `ReportStatsd`
+func ReportStatsd() {
+	client := getClient()
+	if client == nil {
+		return
+	}
```

Comment:
that's seems like a bug rather than an edge case
we call `ReportStatsd()` only after setting a client
and if we call `ReportStatsd()` without setting a client that's a bug
so I don't understand checking that

Reply:
> we call ReportStatsd() only after setting a client
That's an assumption based on the current state of things, but nothing prevents someone from misusing the library.
Generally I'd prefer things to fail fast and loudly, but given that this is an _instrumentation_ library, I'd rather be more play on the safer side here and ensure that we won't panic or anything like that.
Perhaps a middle ground here is adding a log entry to signal the library client that something is wrong?